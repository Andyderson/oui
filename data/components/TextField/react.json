{"description":"Generates an `input` element (optionally wrapped in a label) and accepts\nmost of the common input types.","displayName":"Input","methods":[{"name":"renderInput","docblock":null,"modifiers":[],"params":[{"name":"{\n  note,\n  defaultValue,\n  displayError = false,\n  isDisabled = false,\n  isFilter = false,\n  isOptional,\n  isReadOnly,\n  onBlur,\n  isRequired,\n  label,\n  min,\n  max,\n  onChange,\n  onInput,\n  onKeyDown,\n  onFocus,\n  placeholder,\n  testSection,\n  type,\n  value,\n}","type":null}],"returns":null}],"props":{"defaultValue":{"type":{"name":"string"},"required":false,"description":"The default value of the input used on initial render"},"displayError":{"type":{"name":"bool"},"required":false,"description":"Toggle error state styles"},"isDisabled":{"type":{"name":"bool"},"required":false,"description":"Prevents input from being modified and appears disabled"},"isFilter":{"type":{"name":"bool"},"required":false,"description":"Includes error if true"},"isOptional":{"type":{"name":"bool"},"required":false,"description":"Adds an optional label if there is a label provided\n @param {Object} props Object of props\n @returns {Error} Error or null"},"isReadOnly":{"type":{"name":"bool"},"required":false,"description":"Prevents input from being modified but doesn't appear disabled"},"isRequired":{"type":{"name":"bool"},"required":false,"description":"Prevents input from being submitted without value"},"label":{"type":{"name":"string"},"required":false,"description":"Text that describes the input"},"max":{"type":{"name":"number"},"required":false,"description":"Max value for the `input`. Should be used only when `type` is `number`."},"min":{"type":{"name":"number"},"required":false,"description":"Min value for the `input`. Should be used only when `type` is `number`."},"note":{"type":{"name":"string"},"required":false,"description":"Append note near form input."},"onBlur":{"type":{"name":"func"},"required":false,"description":"Function that fires when the input loses focus. It fires regardless of\nwhether the value has changed."},"onChange":{"type":{"name":"func"},"required":false,"description":"Function that fires when the input loses focus after the value changes"},"onFocus":{"type":{"name":"func"},"required":false,"description":"Function that fires when the input gains focus"},"onInput":{"type":{"name":"func"},"required":false,"description":"Function that fires on keypress"},"onKeyDown":{"type":{"name":"func"},"required":false,"description":"Function that fires when a key is pressed down"},"placeholder":{"type":{"name":"string"},"required":false,"description":"Input placeholder text"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"},"type":{"type":{"name":"enum","value":[{"value":"'text'","computed":false},{"value":"'password'","computed":false},{"value":"'date'","computed":false},{"value":"'number'","computed":false},{"value":"'email'","computed":false},{"value":"'url'","computed":false},{"value":"'search'","computed":false},{"value":"'tel'","computed":false}]},"required":true,"description":"Supported input types"},"value":{"type":{"name":"string"},"required":false,"description":"Text within the input"}},"private":false}