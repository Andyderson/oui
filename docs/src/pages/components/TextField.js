
  import React from 'react';
  import ComponentContent from '../../layouts/templates/ComponentContent';

   import data from '../../../../data/components/TextField/react.json'; import examples from '../../../../src/components/TextField/example'; import readme from 'raw-loader!../../../../src/components/TextField/README.md';

  let dataObject = {
    react: data ? data : null, 
    examples: examples ? examples : null,
    readme: readme ? readme : null
  }
  
  class TextFieldComponent extends React.Component {
  
    render() {
      return (
        <ComponentContent data={ dataObject } />
      );
    };
  }
  
  export default TextFieldComponent
