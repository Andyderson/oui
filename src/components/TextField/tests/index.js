import React from 'react';
import TextField from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/TextField', () => {
  it('should render a simple input with type text', () => {
    const output = shallow(
      <TextField
        type='text'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render with a simple label', () => {
    const output = shallow(
      <TextField
        label='simple label'
        type='text'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render with a simple label and a note', () => {
    const output = shallow(
      <TextField
        label='Simple label'
        note='simple note'
        type='text'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render error style', () => {
    const output = shallow(
      <TextField
        displayError={ true }
        label='Simple label'
        note='simple note'
        type='text'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render with a placeholder', () => {
    const output = shallow(
      <TextField
        placeholder='Simple placeholder'
        type='text'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render search icon', () => {
    const output = shallow(
      <TextField
        isFilter={ true }
        type='text'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render max and min limits', () => {
    const output = shallow(
      <TextField
        min={ 0 }
        max={ 100 }
        type='number'
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
