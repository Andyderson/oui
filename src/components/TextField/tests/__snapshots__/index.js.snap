// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TextField should render HTML type=number for when prop.type is percent 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input text--right"
      disabled={false}
      type="number"
    />
    <span
      className="push-half--left push--right"
    >
      %
    </span>
  </span>
</div>
`;

exports[`components/TextField should render a simple input with type text 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input"
      disabled={false}
      type="text"
    />
  </span>
</div>
`;

exports[`components/TextField should render error style 1`] = `
<div
  className="oui-form-bad-news"
  data-oui-component={true}
>
  <Label
    displayError={true}
  >
    Simple label
  </Label>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input"
      disabled={false}
      type="text"
    />
  </span>
  <div
    className="oui-form-note"
  >
    simple note
  </div>
</div>
`;

exports[`components/TextField should render max and min limits 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input text--right"
      disabled={false}
      max={100}
      min={0}
      type="number"
    />
  </span>
</div>
`;

exports[`components/TextField should render search icon 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input oui-text-input--search"
      disabled={false}
      type="text"
    />
  </span>
</div>
`;

exports[`components/TextField should render with a placeholder 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input"
      disabled={false}
      placeholder="Simple placeholder"
      type="text"
    />
  </span>
</div>
`;

exports[`components/TextField should render with a simple label 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <Label
    displayError={false}
  >
    simple label
  </Label>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input"
      disabled={false}
      type="text"
    />
  </span>
</div>
`;

exports[`components/TextField should render with a simple label and a note 1`] = `
<div
  className={null}
  data-oui-component={true}
>
  <Label
    displayError={false}
  >
    Simple label
  </Label>
  <span
    className="flex flex-align--center"
  >
    <input
      className="oui-text-input"
      disabled={false}
      type="text"
    />
  </span>
  <div
    className="oui-form-note"
  >
    simple note
  </div>
</div>
`;
